en/docs/learn/api-security/oauth2/grant-types/password-grant.md

```diff
@@ -2,20 +2,30 @@
 
+The resource owner password credentials grant type is suitable in cases where the resource owner has a trust relationship with the client (e.g., a service’s own mobile client) and in situations where the client can obtain the resource owner’s credentials.
+
+#### The flow
+
+Instead of redirecting the user to the authorization server, the client itself will ask the user for the resource owner's username and password. The client will then send these credentials to the authorization server along with the client’s own credentials.
+
+The diagram below illustrates the resource owner password credentials grant flow.
+
+![]({{base_path}}/assets/img/learn/oauth-resource-owner-diagram.png)
+
 #### Prerequisites
 
--   A valid user account in the API Developer Portal. You can self sign up if it is [enabled by an admin](../../../../customizations/CustomizingTheAPIStore/enabling-or-disabling-self-signup) .
+-   A valid user account in the API Developer Portal. You can self sign up if it is [enabled by an admin]({{base_path}}/learn/consume-api/customizations/customizing-the-developer-portal/enabling-or-disabling-self-signup/).
 -   A valid consumer key and consumer secret pair. Initially, these keys must be generated through the API Developer Portal by clicking **GENERATE KEYS** on the **Production Keys** tab of the application.
--   A running API Gateway instance (typically an API Manager instance should be running). For instructions on API Gateway, see [Components](../../../../../../getting-started/overview/#api-gateway).
+-   A running API Gateway instance (typically an API Manager instance should be running). For instructions on API Gateway, see [Components]({{base_path}}/getting-started/overview/#api-gateway).
 
--   If the Key Manager is on a different server than the API Gateway, change the server URL (host and ports) of the Key Manager accordingly by adding following configuration in `<AM_HOME>/repository/conf/deployment.toml` file.
+-   If the Key Manager is on a different server than the API Gateway, change the server URL (host and ports) of the Key Manager accordingly by adding following configuration in `<APIM_HOME>/repository/conf/deployment.toml` file.
 
 ``` java
-    [apim.key_manager]
-    configuration.ServerURL = "<key-manager-server-url>"
+[apim.key_manager]
+configuration.ServerURL = "<key-manager-server-url>"
 
 ```
--   If you have multiple Carbon servers running on the same computer, [change the port with an offset](../../../../../../reference/guides/changing-the-default-ports-with-offset/) to avoid port conflicts.
+If you have multiple Carbon servers running on the same computer, [change the port with an offset]({{base_path}}/install-and-setup/deploying-wso2-api-manager/changing-the-default-ports-with-offset/#changing-the-default-ports-with-offset) to avoid port conflicts.
```

Comment:
Is port offsetting step required here?

Reply:
Not necessary, but if there are multiple carbon servers then we need to follow this step.